pkg/repositories/transformers/task_execution.go

```diff
@@ -209,6 +211,9 @@ func UpdateTaskExecutionModel(request *admin.TaskExecutionEventRequest, taskExec
 	taskExecutionClosure.Phase = request.Event.Phase
 	taskExecutionClosure.UpdatedAt = request.Event.OccurredAt
 	taskExecutionClosure.Logs = mergeLogs(taskExecutionClosure.Logs, request.Event.Logs)
+	if len(request.Event.Reason) > 0 {
+		taskExecutionClosure.Reason = request.Event.Reason
```

Comment:
Can we add a unit test to verify this gets added in the closure from the event along with the task type

Reply:
yup already done as part of this PR